### ISSUE TITLE
Sort out logging in tests

### ISSUE THREAD

**Issue by jroper** [playframework/playframework#1717]
The tests dump a massive amount of unnecessary log statements.  We should sort that out.

**Comment by EECOLOR** [comment]
I have the following output, this does not indicate any source as a problem:

``` scala
22:02:06,259 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback.groovy]
22:02:06,260 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
22:02:06,260 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback.xml] at [jar:file:/home/eecolor/.ivy2/cache/com.typesafe.play/play_2.11/jars/play_2.11-2.3.7.jar!/logback.xml]
22:02:06,275 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@31bd1a7 - URL [jar:file:/home/eecolor/.ivy2/cache/com.typesafe.play/play_2.11/jars/play_2.11-2.3.7.jar!/logback.xml] is not of type file
22:02:06,654 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - debug attribute not set
22:02:06,661 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
22:02:06,690 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [STDOUT]
22:02:06,742 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
22:02:06,868 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to ERROR
22:02:06,868 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[ROOT]
22:02:06,869 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
22:02:06,875 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@17612c89 - Registering current configuration as safe fallback point
22:02:07,081 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@6cb2e25c - URL [jar:file:/home/eecolor/.ivy2/cache/com.typesafe.play/play_2.11/jars/play_2.11-2.3.7.jar!/logger.xml] is not of type file
22:02:07,081 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@542b3dc4 - URL [jar:file:/home/eecolor/.ivy2/cache/com.typesafe.play/play_2.11/jars/play_2.11-2.3.7.jar!/logger.xml] is not of type file
22:02:07,084 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - debug attribute not set
22:02:07,084 |-INFO in ch.qos.logback.core.joran.action.ConversionRuleAction - registering conversion word coloredLevel with class [play.api.Logger$ColoredLevel]
22:02:07,084 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.FileAppender]
22:02:07,086 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [FILE]
22:02:07,094 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
22:02:07,095 |-INFO in ch.qos.logback.core.FileAppender[FILE] - File property is set to [/home/eecolor/rhinofly/play-s3/test/logs/application.log]
22:02:07,097 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
22:02:07,097 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [STDOUT]
22:02:07,097 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
22:02:07,098 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [play] to INFO
22:02:07,099 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [application] to DEBUG
22:02:07,099 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.avaje.ebean.config.PropertyMapLoader] to OFF
22:02:07,099 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.avaje.ebeaninternal.server.core.XmlConfigLoader] to OFF
22:02:07,099 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.avaje.ebeaninternal.server.lib.BackgroundThread] to OFF
22:02:07,099 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.gargoylesoftware.htmlunit.javascript] to OFF
22:02:07,099 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to ERROR
22:02:07,101 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[ROOT]
22:02:07,101 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [FILE] to Logger[ROOT]
22:02:07,101 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
22:02:07,101 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@4cdc40f2 - Registering current configuration as safe fallback point
22:02:07,101 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - debug attribute not set
22:02:07,101 |-INFO in ch.qos.logback.core.joran.action.ConversionRuleAction - registering conversion word coloredLevel with class [play.api.Logger$ColoredLevel]
22:02:07,101 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.FileAppender]
22:02:07,102 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [FILE]
22:02:07,103 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
22:02:07,107 |-INFO in ch.qos.logback.core.FileAppender[FILE] - File property is set to [/home/eecolor/rhinofly/play-s3/test/logs/application.log]
22:02:07,107 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
22:02:07,107 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [STDOUT]
22:02:07,108 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
22:02:07,108 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [play] to INFO
22:02:07,108 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [application] to DEBUG
22:02:07,108 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.avaje.ebean.config.PropertyMapLoader] to OFF
22:02:07,108 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.avaje.ebeaninternal.server.core.XmlConfigLoader] to OFF
22:02:07,108 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.avaje.ebeaninternal.server.lib.BackgroundThread] to OFF
22:02:07,109 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.gargoylesoftware.htmlunit.javascript] to OFF
22:02:07,109 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to ERROR
22:02:07,109 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[ROOT]
22:02:07,109 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [FILE] to Logger[ROOT]
22:02:07,109 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
22:02:07,109 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@102bf74a - Registering current configuration as safe fallback point
22:02:07,116 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@36782391 - URL [jar:file:/home/eecolor/.ivy2/cache/com.typesafe.play/play_2.11/jars/play_2.11-2.3.7.jar!/logger.xml] is not of type file
22:02:07,121 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - debug attribute not set
22:02:07,122 |-INFO in ch.qos.logback.core.joran.action.ConversionRuleAction - registering conversion word coloredLevel with class [play.api.Logger$ColoredLevel]
22:02:07,122 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.FileAppender]
22:02:07,122 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [FILE]
22:02:07,122 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
22:02:07,123 |-INFO in ch.qos.logback.core.FileAppender[FILE] - File property is set to [/home/eecolor/rhinofly/play-s3/test/logs/application.log]
22:02:07,124 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
22:02:07,124 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [STDOUT]
22:02:07,124 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
22:02:07,126 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [play] to INFO
22:02:07,126 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [application] to DEBUG
22:02:07,126 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.avaje.ebean.config.PropertyMapLoader] to OFF
22:02:07,126 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.avaje.ebeaninternal.server.core.XmlConfigLoader] to OFF
22:02:07,127 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.avaje.ebeaninternal.server.lib.BackgroundThread] to OFF
22:02:07,127 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.gargoylesoftware.htmlunit.javascript] to OFF
22:02:07,128 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to ERROR
22:02:07,128 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[ROOT]
22:02:07,128 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [FILE] to Logger[ROOT]
22:02:07,128 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
22:02:07,128 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@9672087 - Registering current configuration as safe fallback point
22:02:07,149 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@518883e3 - URL [jar:file:/home/eecolor/.ivy2/cache/com.typesafe.play/play_2.11/jars/play_2.11-2.3.7.jar!/logger.xml] is not of type file
22:02:07,152 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - debug attribute not set
22:02:07,153 |-INFO in ch.qos.logback.core.joran.action.ConversionRuleAction - registering conversion word coloredLevel with class [play.api.Logger$ColoredLevel]
22:02:07,153 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.FileAppender]
22:02:07,153 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [FILE]
22:02:07,155 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
22:02:07,156 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@598ad5e6 - URL [jar:file:/home/eecolor/.ivy2/cache/com.typesafe.play/play_2.11/jars/play_2.11-2.3.7.jar!/logger.xml] is not of type file
22:02:07,157 |-INFO in ch.qos.logback.core.FileAppender[FILE] - File property is set to [/home/eecolor/rhinofly/play-s3/test/logs/application.log]
22:02:07,157 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
22:02:07,157 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [STDOUT]
22:02:07,157 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
22:02:07,159 |-ERROR in ch.qos.logback.core.pattern.parser.Compiler@29e11cf7 - There is no conversion class registered for conversion word [coloredLevel]
22:02:07,159 |-ERROR in ch.qos.logback.core.pattern.parser.Compiler@29e11cf7 - [coloredLevel] is not a valid conversion word
22:02:07,159 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [play] to INFO
22:02:07,161 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [application] to DEBUG
22:02:07,161 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.avaje.ebean.config.PropertyMapLoader] to OFF
22:02:07,161 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.avaje.ebeaninternal.server.core.XmlConfigLoader] to OFF
22:02:07,161 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.avaje.ebeaninternal.server.lib.BackgroundThread] to OFF
22:02:07,161 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [com.gargoylesoftware.htmlunit.javascript] to OFF
22:02:07,161 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to ERROR
22:02:07,164 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[ROOT]
22:02:07,164 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [FILE] to Logger[ROOT]
22:02:07,164 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
22:02:07,164 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - debug attribute not set
22:02:07,165 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@7a34b19b - Registering current configuration as safe fallback point
22:02:07,165 |-INFO in ch.qos.logback.core.joran.action.ConversionRuleAction - registering conversion word coloredLevel with class [play.api.Logger$ColoredLevel]
22:02:07,165 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.FileAppender]
22:02:07,165 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [FILE]
```

**Comment by EECOLOR** [comment]
Logback will log all these things if it encounters an error. In the above case it's:

    22:02:07,159 |-ERROR in ch.qos.logback.core.pattern.parser.Compiler@29e11cf7 - There is no conversion class registered for conversion word [coloredLevel]
    22:02:07,159 |-ERROR in ch.qos.logback.core.pattern.parser.Compiler@29e11cf7 - [coloredLevel] is not a valid conversion word

There must be some classloader problem somewhere, but I am not sure where.

**Comment by jroper** [comment]
I'm guessing then that logback loads classes from the context classloader, and sometimes we don't have it set properly when we initialise it.  Shouldn't be too hard to fix.

**Comment by jroper** [comment]
Fixed in #3941.
